### DISCUSSION TITLE
Foxit PDF Reader - Dectection Script fails on x64

### DISCUSSION THREAD

**Discussion by verinco-davy** [svrooij/WinTuner#70] [Q&A]
Hello

when installing Foxit PDF Reader (Foxit.FoxitReader) the detection fails because it cann't find WinGet.

I think it is because the detection script is run as 32 bits on 64 bits clients. (tested & can conform it)

Can this be adjusted with a switch? Or something i'm overlooking?

thx!

**Comment by verinco-davy** [comment]
Mozilla Firefox (Mozilla.Firefox) just the same

**Comment by svrooij** [comment]
If I build it correctly, it should write a detection log, can you paste the output here?

I thought that the detections scripts would run on 64-bit as well 😒

```ps1
Start-Transcript -Path ""$env:ProgramData\Microsoft\IntuneManagementExtension\Logs\$packageId-detection.log"" -Force
```

**Reply by svrooij** [reply]
I removed your last message, in the email response was a lot of garbage and no image.

**Reply by svrooij** [reply]
Here is the detection script, any ideas on how I would solve this?
Can I just switch the detections scripts to x64?

https://github.com/svrooij/WingetIntune/blob/20a1cf8c2112c10fe8037deb34caab6e9adf0abf/src/WingetIntune/Intune/IntuneManagerConstants.cs#L52-L112

**Reply by verinco-davy** [reply]
Hello

i guess so; if i switch this to NO; it works:

![2024-05-23 14_11_58-Toepassing bewerken - Microsoft Intune-beheercentrum - Brave](https://github.com/svrooij/WingetIntune/assets/83293240/9a599bd1-f988-4314-b178-cdb47ebd4618)

**Reply by svrooij** [reply]
I thought I was certain it did that already, or did you manually upload the app?

**Reply by verinco-davy** [reply]
No, through a script i'm making 

command is this

New-WtWingetPackage -PackageId $WinGetProgram -PackageFolder "C:\WinGetPackages" | Deploy-WtWin32App -Username $MSUsername

**Reply by svrooij** [reply]
I found the issue:

1. both packages have multiple installers, if you don't specify the installer architecture you want it will pick X86
2. the architecture is used to check if it should run the detection script at 32-bit

Solution for now, until I make it always run on 64-bit is to package the 64-bit version with the `-Architecture X64` paramater for the `New-WtWingetPackage` command.

**Reply by svrooij** [reply]
Version 0.8.7 should fix your issue
